File: 송하연/9월 5주차/야근지수.java
import java.util.*;

class Solution {
    public long solution(int n, int[] works) {
        long answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        
        for(int time:works){
            pq.add(time);
        }
        //System.out.println(pq);
        
        for(int i=0; i<n; i++){
            if(pq.isEmpty()) return 0;
            
            int max = pq.poll();
            int minus = max-1;
            if(minus!=0) pq.add(max-1);
        }
        
        long result =0;
        while(!pq.isEmpty()){
           result += Math.pow(pq.poll(),2);
        }

        return answer=result;
    }
}

File: 한승우/9월 5주차/야근 지수.cpp
#include <string>
#include <vector>
#include <queue>
#include <cmath>

using namespace std;

long long solution(int n, vector<int> works) {
    long long answer = 0;
    
    priority_queue<int> pq(works.begin(), works.end());
    
    while(n > 0){
        int num = pq.top();
        if(num == 0) break;
        
        pq.pop();
        num--;
        n--;
        pq.push(num);
    }
    
    while(!pq.empty()){
        answer += pow(pq.top(), 2);
        pq.pop();
    }
    
    return answer;
}